src/service/action-impl.js

```diff
@@ -640,14 +661,56 @@ export class ActionService {
     let actionMap = node[ACTION_MAP_];
     if (actionMap === undefined) {
       actionMap = null;
-      if (node.hasAttribute('on')) {
-        actionMap = parseActionMap(node.getAttribute('on'), node);
+      const hasAction = node.hasAttribute('on');
+      if (hasAction) {
+        const action = node.getAttribute('on');
+        const actionMacroDef = this.getActionMacroDef_(action);
+
+        actionMap = parseActionMap(
+            !!actionMacroDef ? actionMacroDef.action : action,
+            node, actionMacroDef ? actionMacroDef.actionInfoDef.args : null);
+
+        node[ACTION_MAP_] = actionMap;
       }
-      node[ACTION_MAP_] = actionMap;
     }
     return actionMap;
   }
 
+  /**
+   * Check if an action call references an action macro and if so returns
+   * the details of that action macro.
+   * @param {string} actionCall
+   * @private
+   * @return {?ActionMacroDef}
+   */
+  getActionMacroDef_(actionCall) {
+    // An action macro has the following formats:
+    //   event:action-macro-id()
+    //   event.action-macro-id
+    //   event.action-macro-id(arg1=1, arg2=2)
+    const actionMacroCall = /^\w+:(\w+[a-zA-Z'-]+)((\(.*\))?)$/;
```

Comment:
Is `[a-zA-Z'-]+` right? What are valid characters for `id` attribute?

More broadly, I'm curious why we can't leverage the existing parsing code in `parseActionMap()` below.